### PR TITLE
[Docs] Migrate using-preact example to typescript

### PR DESCRIPTION
**Pull request by HaNdTriX**
## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/using-preact/package.json [hunk 1]
```diff
@@ -5,14 +5,17 @@
...
+    "next": "latest",
+    "next-plugin-preact": "latest",
+    "preact": "^10.10.6",
+    "preact-render-to-string": "^5.2.3",
+    "react": "npm:@preact/compat@^17.1.1",
+    "react-dom": "npm:@preact/compat@^17.1.1",
+    "react-ssr-prepass": "npm:preact-ssr-prepass@1.2.0"
+  },
+  "devDependencies": {
+    "@types/node": "18.7.15",
```
Comment (balazsorban44) [inline examples/using-preact/package.json:18]: `@types/react` should be added as well.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Minor comment, looks good otherwise. :+1:

File: examples/using-preact/package.json (hunk 1 above)
Comment (HaNdTriX) [reply examples/using-preact/package.json:18]: Sure? I thought preact has its own types.

File: examples/using-preact/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "forceConsistentCasingInFileNames": true,
+    "noEmit": true,
+    "incremental": true,
+    "esModuleInterop": true,
+    "moduleResolution": "node",
+    "module": "esnext",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "jsx": "preserve",
+    "jsxFactory": "h",
```
Comment (HaNdTriX) [inline examples/using-preact/tsconfig.json:17]: This should do the trick

**Review by ijjk — APPROVED** [review]

**Pull request merged**